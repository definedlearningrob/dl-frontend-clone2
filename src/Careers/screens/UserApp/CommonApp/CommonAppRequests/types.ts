// TYPES
import { COMMON_APP_FORM_STATUS } from '@dc/resources/enums';

export type CommonAppVariables = {
  page: number;
  perPage: number;
};

export type RecommendationRequestVariables = {
  studentUuid: string;
};

export type CommonAppRequestsData = {
  recommendationRequests: RecommendationRequestsData;
};

export type CommonAppRequestData = {
  recommendationRequest: RecommendationRequestWithForms;
};

export type CommonAppForm = {
  formType: COMMON_APP_FORM_TYPES;
  status: COMMON_APP_FORM_STATUS;
  previewUrl?: string;
  deadline: string | null;
};

export type RecommendationRequestsData = {
  nodes: RecommendationRequest[];
  nodesCount: number;
  pagesCount: number;
};

export type RecommendationRequest = {
  deadline: string | null;
  applicant: Applicant;
  submittedFormsCount: number;
  totalFormsCount: number;
};

export type RecommendationRequestWithForms = {
  applicant: Applicant;
  forms: CommonAppForm[];
};

export type Applicant = {
  applicantId: string;
  email: string;
  firstName: string;
  lastName: string;
  uuid: string;
};

// ENUMS
export enum RECOMMENDATION_REQUEST_STATUSES {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
}

export enum RECOMMENDER_TYPES {
  TEACHER = 'TEACHER',
  COUNSELOR = 'COUNSELOR',
}

export enum COMMON_APP_FORM_TYPES {
  COUNSELOR_EARLY_DECISION_2 = 'COUNSELOR_EARLY_DECISION_2',
  COUNSELOR_EARLY_DECISION = 'COUNSELOR_EARLY_DECISION',
  COUNSELOR_FEE_WAIVER = 'COUNSELOR_FEE_WAIVER',
  COUNSELOR_FINAL_REPORT = 'COUNSELOR_FINAL_REPORT',
  COUNSELOR_MIDYEAR_REPORT = 'COUNSELOR_MIDYEAR_REPORT',
  COUNSELOR_OPTIONAL_REPORT_2 = 'COUNSELOR_OPTIONAL_REPORT_2',
  COUNSELOR_OPTIONAL_REPORT = 'COUNSELOR_OPTIONAL_REPORT',
  COUNSELOR_PROFILE = 'COUNSELOR_PROFILE',
  COUNSELOR_RECOMMENDATION = 'COUNSELOR_RECOMMENDATION',
  COUNSELOR_SECONDARY_REPORT = 'COUNSELOR_SECONDARY_REPORT',
  TEACHER_PROFILE = 'TEACHER_PROFILE',
  TEACHER_RECOMMENDATION = 'TEACHER_RECOMMENDATION',
  TEACHER_EVALUATION = 'TEACHER_EVALUATION',
}
