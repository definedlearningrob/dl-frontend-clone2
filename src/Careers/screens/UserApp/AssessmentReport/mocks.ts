import { MockedResponse } from '@apollo/client/testing';

import {
  ASSESSMENT_REPORT,
  TAssessmentReportData,
} from '@dc/graphql/user/queries/reportsAssessmentReport';
import {
  ASSESSMENT_REPORT_FILTERS,
  TAssessmentReportFiltersData,
} from '@dc/graphql/user/queries/assessmentReportFilters';

export const assessmentReportMock: MockedResponse<TAssessmentReportData> = {
  request: {
    query: ASSESSMENT_REPORT,
    variables: {
      filter: { schoolYear: 2023 },
    },
  },
  result: {
    data: {
      reports: {
        assessmentReport: {
          studentsCount: 300,
          summary: {
            assessmentCompleted: 100,
            assessmentTaken: 200,
          },
          highSchoolInterests: [
            { category: 'Artistic', score: 1 },
            { category: 'Conventional', score: 4 },
            { category: 'Enterprising', score: 4 },
            { category: 'Investigative', score: 6 },
            { category: 'Realistic', score: 6 },
            { category: 'Social', score: 9 },
          ],
          middleSchoolInterests: [
            { category: 'Creator', score: 2 },
            { category: 'Doer', score: 3 },
            { category: 'Helper', score: 4 },
            { category: 'Organizer', score: 4 },
            { category: 'Persuader', score: 6 },
            { category: 'Thinker', score: 8 },
          ],
          middleSchoolWorkValues: [
            { category: 'Achievement', averageTokens: 10 },
            { category: 'Independence', averageTokens: 14 },
            { category: 'Recognition', averageTokens: 4 },
            { category: 'Relationships', averageTokens: 6 },
            { category: 'Support', averageTokens: 28 },
            { category: 'Working Conditions', averageTokens: 15 },
          ],
          highSchoolWorkValues: [
            { category: 'Achievement', averageTokens: 20 },
            { category: 'Independence', averageTokens: 10 },
            { category: 'Recognition', averageTokens: 7 },
            { category: 'Relationships', averageTokens: 30 },
            { category: 'Support', averageTokens: 6 },
            { category: 'Working Conditions', averageTokens: 1 },
          ],
          highSchoolStudyPreferences: [
            {
              area: 'Math',
              results: {
                position1: 6,
                position2: 0,
                position3: 0,
                position4: 2,
                position5: 0,
                position6: 0,
                position7: 0,
              },
            },
            {
              area: 'Science',
              results: {
                position1: 0,
                position2: 0,
                position3: 0,
                position4: 0,
                position5: 0,
                position6: 0,
                position7: 8,
              },
            },
            {
              area: 'Social Studies',
              results: {
                position1: 2,
                position2: 5,
                position3: 0,
                position4: 1,
                position5: 0,
                position6: 0,
                position7: 0,
              },
            },
            {
              area: 'Language Arts',
              results: {
                position1: 0,
                position2: 0,
                position3: 2,
                position4: 0,
                position5: 0,
                position6: 6,
                position7: 0,
              },
            },
            {
              area: 'Computers',
              results: {
                position1: 0,
                position2: 1,
                position3: 5,
                position4: 0,
                position5: 1,
                position6: 1,
                position7: 0,
              },
            },
            {
              area: 'Creative Arts',
              results: {
                position1: 0,
                position2: 1,
                position3: 0,
                position4: 0,
                position5: 6,
                position6: 1,
                position7: 0,
              },
            },
            {
              area: 'Trades',
              results: {
                position1: 0,
                position2: 1,
                position3: 1,
                position4: 5,
                position5: 1,
                position6: 0,
                position7: 0,
              },
            },
          ],
          middleSchoolStudyPreferences: [
            {
              area: 'Language Arts',
              results: {
                position1: 0,
                position2: 1,
                position3: 0,
                position4: 1,
                position5: 0,
              },
            },
            {
              area: 'Math',
              results: {
                position1: 0,
                position2: 0,
                position3: 2,
                position4: 0,
                position5: 0,
              },
            },
            {
              area: 'Science',
              results: {
                position1: 0,
                position2: 0,
                position3: 0,
                position4: 0,
                position5: 2,
              },
            },
            {
              area: 'Social Studies',
              results: {
                position1: 2,
                position2: 0,
                position3: 0,
                position4: 0,
                position5: 0,
              },
            },
            {
              area: 'Creative Arts',
              results: {
                position1: 0,
                position2: 1,
                position3: 0,
                position4: 1,
                position5: 0,
              },
            },
          ],
          clusterRecommendationCounts: [
            {
              cluster: {
                name: 'Education & Training',
                id: '3',
                pathways: [
                  {
                    name: 'Administration and Administrative Support',
                    id: '14',
                  },
                  {
                    name: 'Professional Support Services',
                    id: '15',
                  },
                  {
                    name: 'Teaching/Training',
                    id: '16',
                  },
                ],
              },
              pathwayRecommendationCounts: [
                {
                  recommendationsCount: 2,
                  pathway: {
                    name: 'Teaching/Training',
                    id: '16',
                  },
                },
                {
                  recommendationsCount: 1,
                  pathway: {
                    name: 'Administration and Administrative Support',
                    id: '14',
                  },
                },
                {
                  recommendationsCount: 1,
                  pathway: {
                    name: 'Professional Support Services',
                    id: '15',
                  },
                },
              ],
              recommendationsCount: 4,
            },
            {
              cluster: {
                name: 'Business Management & Administration',
                id: '13',
                pathways: [
                  {
                    name: 'General Management',
                    id: '61',
                  },
                  {
                    name: 'Business Information Management',
                    id: '62',
                  },
                  {
                    name: 'Human Resources Management',
                    id: '63',
                  },
                  {
                    name: 'Operations Management',
                    id: '64',
                  },
                  {
                    name: 'Administrative Support',
                    id: '65',
                  },
                ],
              },
              pathwayRecommendationCounts: [
                {
                  recommendationsCount: 3,
                  pathway: {
                    name: 'Administrative Support',
                    id: '65',
                  },
                },
                {
                  recommendationsCount: 1,
                  pathway: {
                    name: 'Business Information Management',
                    id: '62',
                  },
                },
              ],
              recommendationsCount: 4,
            },
            {
              cluster: {
                name: 'Manufacturing',
                id: '9',
                pathways: [
                  {
                    name: 'Production',
                    id: '41',
                  },
                  {
                    name: 'Manufacturing Production Process Development',
                    id: '42',
                  },
                  {
                    name: 'Maintenance, Installation & Repair',
                    id: '43',
                  },
                  {
                    name: 'Quality Assurance',
                    id: '44',
                  },
                  {
                    name: 'Logistics & Inventory Control',
                    id: '45',
                  },
                  {
                    name: 'Health, Safety & Environmental Assurance',
                    id: '46',
                  },
                ],
              },
              pathwayRecommendationCounts: [
                {
                  recommendationsCount: 2,
                  pathway: {
                    name: 'Production',
                    id: '41',
                  },
                },
                {
                  recommendationsCount: 2,
                  pathway: {
                    name: 'Manufacturing Production Process Development',
                    id: '42',
                  },
                },
              ],
              recommendationsCount: 4,
            },
          ],
        },
      },
    },
  },
};

export const assessmentReportFiltersMock: MockedResponse<TAssessmentReportFiltersData> = {
  request: {
    query: ASSESSMENT_REPORT_FILTERS,
    variables: {
      filters: {},
      entityFilter: { nameCont: '' },
      schoolClassFilter: { nameCont: '' },
      userFilter: { fullNameCont: '' },
    },
  },
  result: {
    data: {
      assessmentReportFilters: {
        entities: {
          nodesCount: 1,
          pagesCount: 1,
          nodes: [{ uuid: '1234-5678', name: 'City High School' }],
        },
        gradeLevels: ['10', '11', '12', 'Postsecondary'],
        users: {
          nodesCount: 1,
          pagesCount: 1,
          nodes: [{ uuid: '9876-5432', fullName: 'Anna Cityteach' }],
        },
        schoolClasses: {
          nodesCount: 0,
          pagesCount: 0,
          nodes: [],
        },
      },
    },
  },
};
