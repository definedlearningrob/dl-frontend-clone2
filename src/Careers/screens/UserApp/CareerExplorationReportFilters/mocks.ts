import { MockedResponse } from '@apollo/client/testing';
import {
  CareerExplorationReportFiltersDocument,
  CareerExplorationReportSummaryDocument,
  CareerExplorationReportVisitCountsDocument,
} from '@graphql/dc/users/hooks';

import { TCareerExplorationReportFiltersData } from '@dc/graphql/user/queries/careerExplorationReportFilters';

export const careerExplorationReportSummaryMock = {
  request: {
    query: CareerExplorationReportSummaryDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          summary: {
            engagementsCount: 10,
            clusterSCount: 2,
            pathwaysCount: 5,
            studentCount: 12,
            usersCount: 3,
          },
        },
      },
    },
  },
};

export const visitCountsSpy = jest.fn();

export const careerExplorationReportVisitCountsMock = {
  request: {
    query: CareerExplorationReportVisitCountsDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result() {
    visitCountsSpy();

    return {
      data: {
        reports: {
          pathwayReport: [
            {
              visitsCount: 396,
              cluster: {
                name: 'Agriculture, Food & Natural Resources',
                id: '1',
                __typename: 'Cluster',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '5',
                    name: 'Natural Resources Systems',
                  },
                },
                {
                  visitsCount: 55,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '2',
                    name: 'Animal Systems',
                  },
                },
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '6',
                    name: 'Plant Systems',
                  },
                },
              ],
            },
            {
              visitsCount: 10,
              cluster: {
                name: 'Transportation, Distribution & Logistics',
                id: '11',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 2,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '52',
                    name: 'Facility and Mobile Equipment Maintenance',
                  },
                },
                {
                  visitsCount: 3,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '49',
                    name: 'Transportation Operations',
                  },
                },
                {
                  visitsCount: 5,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '51',
                    name: 'Warehousing & Distribution Center Operations',
                  },
                },
              ],
            },
          ],
        },
      },
    };
  },
};

export const careerExplorationReportFiltersMock: MockedResponse<TCareerExplorationReportFiltersData> =
  {
    request: {
      query: CareerExplorationReportFiltersDocument,
      variables: {
        filters: {},
        entityFilter: { nameCont: '' },
        schoolClassFilter: { nameCont: '' },
        userFilter: { fullNameCont: '' },
      },
    },
    result: {
      data: {
        pathwayReportFilters: {
          entities: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '1234-5678', name: 'City High School' }],
          },
          gradeLevels: ['10', '11', '12', 'Postsecondary'],
          users: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '9876-5432', fullName: 'Anna Cityteach' }],
          },
          schoolClasses: {
            nodesCount: 0,
            pagesCount: 0,
            nodes: [],
          },
        },
      },
    },
  };
