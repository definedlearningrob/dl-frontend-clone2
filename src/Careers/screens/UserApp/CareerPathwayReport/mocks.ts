import { MockedResponse } from '@apollo/client/testing';
import { CareerExplorationReportFullDataDocument } from '@graphql/dc/users/hooks';
import { CareerExplorationReportFullDataQuery } from '@graphql/dc/users/operations';

export const careerExplorationReportMock: MockedResponse<CareerExplorationReportFullDataQuery> = {
  request: {
    query: CareerExplorationReportFullDataDocument,
    variables: {
      filter: { schoolYear: 2023 },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          visitResults: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [
              {
                clusterNames: ['Education & Training'],
                isEnrolled: true,
                pathwayNames: ['Teaching/Training'],
                resourceId: '16',
                resourceName: 'Teaching/Training',
                resourceType: 'Pathway',
                visitorId: '1234',
                visitorName: 'Jeremiah Bell',
                visitorEmail: 'jeremiah.bell@example.com',
                visitorSisId: null,
                visitorType: 'Student',
                visitsCount: 5,
              },
            ],
          },
        },
      },
    },
  },
};
