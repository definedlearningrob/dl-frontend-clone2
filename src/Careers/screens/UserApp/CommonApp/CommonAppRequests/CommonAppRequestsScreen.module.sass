@import '@shared/stylesheets/config.sass'

$max-card-width: 380px
$max-card-width-xxl: 535px
$common-icon-width: 118px

.container
  +flex-container(row, center, center)
  height: calc(100vh - $desktop-header-height)

.skeletonCard
  +flex-container(column)
  height: calc(100% - 40px)

.card
  height: auto !important
  padding: 0 !important
  margin-bottom: $m-md

.cardHeader
  +flex-container(row, space-between, center)
  padding: $m-base $m-base

  +media-breakpoint-up(xxxl)
    padding: $m-md $m-md

.cardTitle
  font-size: $font-size-sm
  margin: 0

  +media-breakpoint-up(xxxl)
    font-size: $font-size-base

.icon
  width: $common-icon-width

.verticalDivider
  border: 1px solid $color-neutral-300
  height: 25%

.rotate
  animation: spin 2s infinite linear
