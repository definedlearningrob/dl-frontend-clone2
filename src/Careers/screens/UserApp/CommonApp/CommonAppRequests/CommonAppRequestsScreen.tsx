import SharedMainContent from '@dc/shared/MainContent/MainContent';
import { useRecommendationRequestsQuery } from '@dc/graphql/user/hooks/useRecommendationRequestsQuery';
import useUserInfo from '@dc/hooks/useUserInfo';
import { TUserInfo } from '@dc/graphql/user/queries/userInfo';
import { CommonAppRequestsList } from '@dc/components/User/CommonAppRequests/CommonAppRequestsList';
import { CommonAppProfileCard } from '@dc/screens/UserApp/CommonApp/CommonAppProfileCard';
import { RECOMMENDER_TYPE } from '@dc/resources/enums';
import { CommonAppRequestSkeleton } from '@dc/screens/UserApp/CommonApp/CommonAppRequestSkeleton/CommonAppRequestSkeleton';
import { CommonAppProfileBar } from '@dc/screens/UserApp/CommonApp/CommonAppProfileBar';
import { useSyncCommonAppData } from '@dc/graphql/shared/hooks/useSyncCommonAppData';
import { RECOMMENDATION_REQUESTS_QUERY } from '@dc/graphql/user/queries/recommendationRequests';
import { CommonAppStarterSkeleton } from '@dc/screens/UserApp/CommonApp/CommonAppRequestSkeleton/CommonAppStarterSkeleton';
import { CommonAppSyncStatus } from '@dc/screens/UserApp/CommonApp/CommonAppSyncStatus/CommonAppSyncStatus';

import SharedCard from '@shared/components/Card/Card';
import LoadingSpinner from '@shared/components/LoadingSpinner/LoadingSpinner';

import styles from './CommonAppRequestsScreen.module.sass';

export const CommonAppRequestsScreen = () => {
  const { data, loading: isRecomendationRequestLoading } = useRecommendationRequestsQuery();

  const {
    userInfo: {
      commonAppData: {
        hasTeacherInvitation,
        hasCounselorInvitation,
        hasCounselorProfileFormCompleted,
        hasTeacherProfileFormCompleted,
      },
    },
  } = useUserInfo<TUserInfo>();

  const { isDataRefreshing, syncCommonAppData, lastSyncedAt } = useSyncCommonAppData([
    RECOMMENDATION_REQUESTS_QUERY,
    // TODO: uncomment after fixing useUserInfo hook
    // userInfoQuery,
  ]);

  const isLoading = isRecomendationRequestLoading && !isDataRefreshing;

  if (isLoading && (hasCounselorProfileFormCompleted || hasTeacherProfileFormCompleted)) {
    return <CommonAppRequestSkeleton />;
  }

  if (isLoading) {
    return (
      <SharedMainContent className={styles.container}>
        <CommonAppStarterSkeleton />
      </SharedMainContent>
    );
  }

  if (!data) {
    return null;
  }

  if (!hasCounselorProfileFormCompleted && !hasTeacherProfileFormCompleted) {
    return (
      <SharedMainContent className={styles.container}>
        {hasCounselorInvitation && <CommonAppProfileCard type={RECOMMENDER_TYPE.COUNSELOR} />}
        {hasTeacherInvitation && hasCounselorInvitation && (
          <div className={styles.verticalDivider} />
        )}
        {hasTeacherInvitation && <CommonAppProfileCard type={RECOMMENDER_TYPE.TEACHER} />}
      </SharedMainContent>
    );
  }

  return (
    <SharedMainContent>
      <CommonAppSyncStatus
        isDataRefreshing={isDataRefreshing}
        lastSyncedAt={lastSyncedAt}
        onSync={syncCommonAppData}
      />
      <SharedCard className={styles.card}>
        <SharedCard.Header className={styles.cardHeader}>
          <CommonAppProfileBar />
        </SharedCard.Header>
      </SharedCard>
      <SharedCard.Body>
        {isRecomendationRequestLoading && (
          <SharedCard className='py-lg'>
            <LoadingSpinner color='primary' size='small' />
          </SharedCard>
        )}
        {!isRecomendationRequestLoading && <CommonAppRequestsList />}
      </SharedCard.Body>
    </SharedMainContent>
  );
};
