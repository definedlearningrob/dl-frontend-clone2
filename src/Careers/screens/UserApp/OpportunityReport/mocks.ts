import { MockedResponse } from '@apollo/client/testing';
import { OpportunityReportFullDataDocument } from '@graphql/dc/users/hooks';
import { OpportunityReportFullDataQuery } from '@graphql/dc/users/operations';
import { ApplicationStatus, OpportunityTypes } from '@graphql/dc/shared/types';
import { ClustersQuery, ClustersQueryVariables } from '@graphql/dc/shared/operations';
import { ClustersDocument } from '@graphql/dc/shared/hooks';

export const clustersMock: MockedResponse<ClustersQuery, ClustersQueryVariables> = {
  request: {
    query: ClustersDocument,
    variables: {},
  },
  result: {
    data: {
      clusters: [
        {
          id: '1',
          name: 'Agriculture, Food & Natural Resources',
          pathways: [
            {
              id: '1',
              name: 'Agribusiness Systems',
            },
            {
              id: '2',
              name: 'Animal Systems',
            },
          ],
        },
      ],
    },
  },
};

export const opportunityResultsMock: MockedResponse<OpportunityReportFullDataQuery> = {
  request: {
    query: OpportunityReportFullDataDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
      resultsFilter: {},
      page: 1,
      perPage: 25,
    },
  },
  result: {
    data: {
      reports: {
        opportunityReport: {
          results: {
            nodesCount: 3,
            pagesCount: 1,
            nodes: [
              {
                applicationDeadline: null,
                applicationStatus: ApplicationStatus.ACCEPTED,
                assignmentsSubmitted: 3,
                assignmentsToSubmit: 6,
                checkInsSubmitted: 0,
                checkInsToSubmit: 0,
                clusterNames: ['Marketing', 'Transportation, Distribution & Logistics'],
                isFavorite: false,
                opportunityName: 'Test',
                opportunityPartnerNames: [],
                opportunityType: OpportunityTypes.VIRTUAL_INTERNSHIP,
                pathwayNames: ['Travel & Tourism', 'Warehousing & Distribution Center Operations'],
                studentGradeLevel: '3',
                studentId: '10458251',
                studentName: 'Amanda Brown',
                studentSisId: '242424',
              },
              {
                applicationDeadline: null,
                applicationStatus: ApplicationStatus.ACCEPTED,
                assignmentsSubmitted: 0,
                assignmentsToSubmit: 0,
                checkInsSubmitted: 0,
                checkInsToSubmit: 0,
                clusterNames: [
                  'Agriculture, Food & Natural Resources',
                  'Architecture & Construction',
                ],
                isFavorite: false,
                opportunityName: 'Test 2',
                opportunityPartnerNames: [],
                opportunityType: OpportunityTypes.VIRTUAL_INTERNSHIP,
                pathwayNames: [
                  'A/V Technology & Film',
                  'Administration and Administrative Support',
                ],
                studentGradeLevel: '7',
                studentId: '10458576',
                studentName: 'Jon James',
                studentSisId: null,
              },
              {
                applicationDeadline: '2055-01-01',
                applicationStatus: ApplicationStatus.ACCEPTED,
                assignmentsSubmitted: null,
                assignmentsToSubmit: null,
                checkInsSubmitted: null,
                checkInsToSubmit: null,
                clusterNames: [
                  'Agriculture, Food & Natural Resources',
                  'Business Management & Administration',
                ],
                isFavorite: true,
                opportunityName: 'molestias',
                opportunityPartnerNames: [],
                opportunityType: OpportunityTypes.PRACTICUM,
                pathwayNames: ['Business Information Management', 'Diagnostic Services'],
                studentGradeLevel: '7',
                studentId: '10458576',
                studentName: 'Jon James',
                studentSisId: null,
              },
            ],
          },
        },
      },
    },
  },
};
