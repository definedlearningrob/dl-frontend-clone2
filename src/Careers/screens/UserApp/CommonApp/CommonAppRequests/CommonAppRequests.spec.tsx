import { MockedProvider } from '@apollo/client/testing';
import { Route } from 'react-router-dom';
import { screen, waitFor } from '@testing-library/react';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { CommonAppRequestsScreen } from '@dc/screens/UserApp/CommonApp/CommonAppRequests/CommonAppRequestsScreen';
import { RECOMMENDATION_REQUESTS_QUERY } from '@dc/graphql/user/queries/recommendationRequests';
import cacheConfig from '@dc/graphql/cacheConfig';
import { userInfoMock } from '@dc/tests/mocks/userMocks';
import { SYNC_STATUS } from '@dc/resources/enums';

const commonAppRecommendationsResponse = {
  data: {
    recommendationRequests: {
      nodesCount: 2,
      pagesCount: 1,
      nodes: [
        {
          applicant: {
            firstName: 'John',
            lastName: 'Doe',
            uuid: '0123',
            applicantId: '1',
            email: 'johndoe@test.org',
          },
          deadline: '',
          submittedFormsCount: 0,
          totalFormsCount: 2,
        },
        {
          applicant: {
            firstName: 'Jane',
            lastName: 'Doe',
            uuid: '0124',
            applicantId: '2',
            email: 'johedoe@test.org',
          },
          deadline: '',
          submittedFormsCount: 1,
          totalFormsCount: 4,
        },
      ],
    },
  },
};

const commonAppRecommendationsMock = {
  request: {
    query: RECOMMENDATION_REQUESTS_QUERY,
    variables: {
      page: 1,
      perPage: 20,
    },
  },
  result: commonAppRecommendationsResponse,
};

const mocks = [commonAppRecommendationsMock];

const syncStatus = {
  lastSyncedAt: '2021-08-09T12:00:00.000Z',
  status: SYNC_STATUS.COMPLETED,
};

const defaultCommonAppData = {
  commonAppData: {
    hasRecommenderInvitation: true,
    hasTeacherInvitation: false,
    hasCounselorInvitation: false,
    hasCounselorProfileFormCompleted: false,
    hasTeacherProfileFormCompleted: false,
    syncStatus,
  },
};

const renderComponentCardWithProvider = (commonAppData = defaultCommonAppData) =>
  renderWithRouterAndReduxProvider(
    <MockedProvider cache={cacheConfig} mocks={mocks}>
      <Route path='/post-secondary/commonapp-requests'>
        <UserInfoProvider
          value={{ userInfo: { ...userInfoMock.result.data.userInfo, ...commonAppData } }}>
          <CommonAppRequestsScreen />
        </UserInfoProvider>
      </Route>
    </MockedProvider>,
    { route: '/post-secondary/commonapp-requests' }
  );

describe('CommonAppRequest counselor and teacher profile forms', () => {
  it('should render skeletons when has only RecommenderInvitation', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: false,
        hasCounselorInvitation: false,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus,
      },
    };

    const { container } = renderComponentCardWithProvider(commonAppData);
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('should render component for fill up counselor profile from when hasCounselorInvitation', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: false,
        hasCounselorInvitation: true,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus,
      },
    };

    renderComponentCardWithProvider(commonAppData);

    const completeCounselorProfile = await screen.findByRole('button', {
      name: 'Complete counselor form',
    });

    expect(completeCounselorProfile).toBeInTheDocument();
  });

  it('should render component for fill up teacher  from when hasTeacherInvitation', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: true,
        hasCounselorInvitation: false,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus,
      },
    };

    renderComponentCardWithProvider(commonAppData);

    const completeTeacherProfile = await screen.findByRole('button', {
      name: 'Complete teacher form',
    });

    expect(completeTeacherProfile).toBeInTheDocument();
  });

  it('should render both components for fillUp from when hasTeacherInvitation and hasCounselorInvitation', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: true,
        hasCounselorInvitation: true,
        hasCounselorProfileFormCompleted: false,
        hasTeacherProfileFormCompleted: false,
        syncStatus,
      },
    };

    renderComponentCardWithProvider(commonAppData);

    const completeCounselorProfile = await screen.findByRole('button', {
      name: 'Complete counselor form',
    });

    const completeTeacherProfile = await screen.findByRole('button', {
      name: 'Complete teacher form',
    });

    expect(completeCounselorProfile).toBeInTheDocument();
    expect(completeTeacherProfile).toBeInTheDocument();
  });

  it('should render data counselor profile is completed', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: true,
        hasCounselorInvitation: true,
        hasCounselorProfileFormCompleted: true,
        hasTeacherProfileFormCompleted: false,
        syncStatus,
      },
    };

    renderComponentCardWithProvider(commonAppData);

    await waitFor(() => {
      const headerElement = screen.getByText('You have completed your counselor profile');
      expect(headerElement).toBeInTheDocument();
    });
  });

  it('should render data teacher profile is completed', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: true,
        hasCounselorInvitation: true,
        hasCounselorProfileFormCompleted: true,
        hasTeacherProfileFormCompleted: false,
        syncStatus,
      },
    };

    renderComponentCardWithProvider(commonAppData);
    const editProfile = await screen.findByRole('link', {
      name: /Edit Profile/i,
    });

    const headerElementCounselor = screen.getByText('You have completed your counselor profile');
    expect(headerElementCounselor).toBeInTheDocument();
    expect(editProfile).toBeInTheDocument();
    expect(editProfile).toHaveAttribute('href', '/forms/counselorprofile');
  });

  it('should render data when teacher and counselor profiles are completed', async () => {
    const commonAppData = {
      commonAppData: {
        hasRecommenderInvitation: true,
        hasTeacherInvitation: true,
        hasCounselorInvitation: true,
        hasCounselorProfileFormCompleted: true,
        hasTeacherProfileFormCompleted: true,
        syncStatus,
      },
    };

    renderComponentCardWithProvider(commonAppData);

    const editProfiles = await screen.findByRole('button', {
      name: 'Edit',
    });

    await waitFor(() => {
      const headerElementCounselor = screen.getByText(
        'You have completed your Counselor and teacher profile.'
      );
      expect(headerElementCounselor).toBeInTheDocument();
      expect(editProfiles).toBeInTheDocument();
    });
  });
});
