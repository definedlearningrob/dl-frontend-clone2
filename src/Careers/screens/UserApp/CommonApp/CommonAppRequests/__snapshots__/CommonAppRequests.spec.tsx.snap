// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonAppRequest counselor and teacher profile forms should render skeletons when has only RecommenderInvitation 1`] = `
<div>
  <div
    class="p-base pt-xs xxxl:p-md xxxl:pt-sm container"
    id="read"
  >
    <div
      class="card contentCard m-base"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="flex mb-base justify-center"
        >
          <img
            src="SvgrURL"
          />
        </div>
        <div
          class="flex flex-col"
        >
          <div
            class="flex w-full justify-center pb-base"
          >
            <div
              class="skeleton-rectangle skeletonRectangle radius-sm height-extra-small width-lg darker animated"
            />
          </div>
          <div
            class="flex w-full justify-center pb-base"
          >
            <div
              class="skeleton-rectangle skeletonRectangle radius-sm height-extra-small width-lg darker animated"
            />
          </div>
          <div
            class="flex basis-1/2 justify-center"
          >
            <div
              class="skeleton-rectangle skeletonRectangle radius-sm height-big width-lg darker animated"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
